packages/block-library/src/navigation/test/use-navigation-menu.js

```diff
@@ -0,0 +1,211 @@
+			hasResolvedCanUserDeleteNavigationMenu: false,
+			hasResolvedCanUserUpdateNavigationMenu: false,
+			hasResolvedNavigationMenus: true,
+			isNavigationMenuMissing: true,
+			isNavigationMenuResolved: false,
+			isResolvingCanUserCreateNavigationMenu: false,
+			isResolvingNavigationMenus: false,
+		} );
+	} );
+
+	it( 'Should also return information about a specific menu when ref is given', () => {
+		setRecords( registry, navigationMenus );
+		expect( useNavigationMenu( 1 ) ).toEqual( {
+			navigationMenu: navigationMenu1,
+			navigationMenus,
+			canSwitchNavigationMenu: true,
+			canUserCreateNavigationMenu: false,
+			canUserDeleteNavigationMenu: false,
+			canUserUpdateNavigationMenu: false,
+			hasResolvedCanUserCreateNavigationMenu: false,
+			hasResolvedCanUserDeleteNavigationMenu: false,
+			hasResolvedCanUserUpdateNavigationMenu: false,
+			hasResolvedNavigationMenus: true,
+			isNavigationMenuMissing: false,
+			isNavigationMenuResolved: false,
+			isResolvingCanUserCreateNavigationMenu: false,
+			isResolvingNavigationMenus: false,
+		} );
+	} );
+
+	it( 'Should return correct permissions (create, update)', () => {
+		setRecords( registry, navigationMenus );
+		setCreatePermission( registry, true );
+		setUpdatePermission( registry, 1, true );
+		expect( useNavigationMenu( 1 ) ).toEqual( {
+			navigationMenu: navigationMenu1,
+			navigationMenus,
+			canSwitchNavigationMenu: true,
+			canUserCreateNavigationMenu: true,
+			canUserDeleteNavigationMenu: false,
+			canUserUpdateNavigationMenu: true,
+			hasResolvedCanUserCreateNavigationMenu: true,
+			hasResolvedCanUserDeleteNavigationMenu: false,
+			hasResolvedCanUserUpdateNavigationMenu: true,
+			hasResolvedNavigationMenus: true,
+			isNavigationMenuMissing: false,
+			isNavigationMenuResolved: false,
+			isResolvingCanUserCreateNavigationMenu: false,
+			isResolvingNavigationMenus: false,
+		} );
+	} );
+
+	it( 'Should return correct permissions (delete only)', () => {
+		setRecords( registry, navigationMenus );
+		setDeletePermission( registry, 1, true );
+		expect( useNavigationMenu( 1 ) ).toEqual( {
+			navigationMenu: navigationMenu1,
+			navigationMenus,
+			canSwitchNavigationMenu: true,
+			canUserCreateNavigationMenu: false,
+			canUserDeleteNavigationMenu: true,
+			canUserUpdateNavigationMenu: false,
+			hasResolvedCanUserCreateNavigationMenu: false,
+			hasResolvedCanUserDeleteNavigationMenu: true,
+			hasResolvedCanUserUpdateNavigationMenu: false,
+			hasResolvedNavigationMenus: true,
+			isNavigationMenuMissing: false,
+			isNavigationMenuResolved: false,
+			isResolvingCanUserCreateNavigationMenu: false,
+			isResolvingNavigationMenus: false,
+		} );
+	} );
+
+	it( 'Should return correct permissions (no permissions)', () => {
+		setRecords( registry, navigationMenus );
+		// Note the ref refers to a different record
+		setDeletePermission( registry, 2, true );
+		expect( useNavigationMenu( 1 ) ).toEqual( {
+			navigationMenu: navigationMenu1,
```

Comment:
```suggestion
		const menuWithNoDeletePermissions = 2;
		const requestedMenu = 1;
		const requestedMenuData = [`navigationMenu${requestedMenu}`];
		
		setRecords( registry, navigationMenus );
		// Note the ref refers to a different record
		setDeletePermission( registry, menuWithNoDeletePermissions, true );
		expect( useNavigationMenu( requestedMenu ) ).toEqual( {
			navigationMenu: requestedMenuData,
```

I guess we could make this more self documenting. It might make the test slightly more complex though so 🤷

Reply:
I like it, thanks Dave!